driver-core/src/main/com/mongodb/TransactionOptions.java

```diff
@@ -61,6 +63,22 @@ public static Builder builder() {
         return new Builder();
     }
 
+    /**
+     * Merge the two provided transaction options, with the first taking precedence over the second.
+     *
+     * @param options the transaction options, which take precedence for any property that is non-null
+     * @param defaultOptions the default transaction options
+     * @return the merged transaction optoins
+     */
+    public static TransactionOptions merge(final TransactionOptions options, final TransactionOptions defaultOptions) {
+        return TransactionOptions.builder()
```

Comment:
I think we still need "notNull" checks here on the options, to prevent NPE at runtime.